torch/csrc/generic/Storage.cpp

```diff
@@ -55,22 +47,28 @@ static PyObject * THPStorage_(pynew)(PyTypeObject *type, PyObject *args, PyObjec
     if (num_args == 0) {
       PyObject *cdata_ptr = PyDict_GetItemString(kwargs, "cdata");
       if (num_kwargs == 1 && cdata_ptr && THPUtils_checkLong(cdata_ptr)) {
-        THWStorage *ptr = (THWStorage*)PyLong_AsVoidPtr(cdata_ptr);
+        c10::StorageImpl *ptr = (c10::StorageImpl*)PyLong_AsVoidPtr(cdata_ptr);
         self->cdata = ptr;
         return (PyObject*)self.release();
       }
     }
     THPUtils_assert(num_kwargs == 0, THPStorageStr "(): invalid keyword arguments");
   }
+  if (allocator == nullptr) {
+#if defined(THC_GENERIC_FILE)
+      allocator = c10::cuda::CUDACachingAllocator::get();
```

Comment:
maybe we can work around this by putting a helper function that returns proper allocator in ATen?

Reply:
I'll try that out